clients/ranchercli/download.go

```diff
@@ -0,0 +1,96 @@
+package ranchercli
+
+import (
+	"fmt"
+	osys "os"
+	"os/exec"
+	"strings"
+
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/sirupsen/logrus"
+)
+
+type Client struct {
+	Session *session.Session
+}
+
+func (c *Client) Create(name string, args ...string) error {
+	c.ExecuteCommand(name, args...)
+
+	return nil
+}
+
+func (c *Client) Exists(rancher, resourceType, name string) (bool, error) {
+	cmdStr := fmt.Sprintf("%s %s ls | grep %s", rancher, resourceType, name)
+	cmd := exec.Command("sh", "-c", cmdStr)
+	err := cmd.Run()
+	if err != nil {
+		if exitError, ok := err.(*exec.ExitError); ok {
+			if exitError.ExitCode() == 1 {
```

Comment:
is there a reason this is only for `1`? If not, can you switch it to `if exitError.ExitCode() != 0`?

Reply:
The `return false, err` on line 35 is capturing other exit codes. Bash commands typically are going to give exit code 1 after running `echo $?` when returning an empty result.

Reply:
I don't get this logic you're still returning false on line 35 why is 1 so special?

Reply:
I mean, `1` just for reasons above. But if both of you think it should just be switched to `!= 0`, it's not a huge deal to me to change it.

Reply:
this is to simplify logic 3 embedded if statements is hard to read.

Reply:
I understand now an exit code of zero shouldn't return false that's why you're checking. I think !=0 would be best. Thanks!

Reply:
Resolved